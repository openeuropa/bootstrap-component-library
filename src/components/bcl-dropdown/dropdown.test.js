import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";
import demoData from "@openeuropa/bcl-data-dropdown/data";

const template = "@oe-bcl/bcl-dropdown/dropdown.html.twig";
const render = (params) => renderTwigFileAsNode(template, params, true);

describe("OE - dropdown", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
  test("renders correctly with dark mode", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, dark: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with trigger link", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, link: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with direction dropup", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, direction: "dropup" })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly without wrapper", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, remove_wrapper: true })
    ).resolves.toMatchSnapshot();
  });

  test("renders correctly with alignment end", () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, alignment: "dropdown-menu-sm-end" })
    ).resolves.toMatchSnapshot();
  });
});
