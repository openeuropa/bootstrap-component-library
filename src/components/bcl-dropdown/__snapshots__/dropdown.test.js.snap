// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - dropdown renders correctly 1`] = `
<jest>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      autocomplete="off"
      class="dropdown-toggle btn btn-primary btn-md"
      data-bs-toggle="dropdown"
      id="dropdown-1"
      type="button"
    >
      Dropdown Toggle
    </button>
    <ul
      aria-labelledby="dropdown-1"
      class="dropdown-menu"
    >
      <li>
        <a
          class="dropdown-item"
          href="#"
        >
          I'm a link
        </a>
      </li>
      <li>
        <button
          class="dropdown-item"
          type="button"
        >
          I'm a button
        </button>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <button
          aria-disabled="true"
          class="dropdown-item"
          disabled="true"
          type="button"
        >
          I'm a disabled button
        </button>
      </li>
      <li>
        <button
          class="dropdown-item active"
          type="button"
        >
          I'm a active button
        </button>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - dropdown renders correctly with alignment end 1`] = `
<jest>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      autocomplete="off"
      class="dropdown-toggle btn btn-primary btn-md"
      data-bs-toggle="dropdown"
      id="dropdown-1"
      type="button"
    >
      Dropdown Toggle
    </button>
    <ul
      aria-labelledby="dropdown-1"
      class="dropdown-menu dropdown-menu-sm-end"
    >
      <li>
        <a
          class="dropdown-item"
          href="#"
        >
          I'm a link
        </a>
      </li>
      <li>
        <button
          class="dropdown-item"
          type="button"
        >
          I'm a button
        </button>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <button
          aria-disabled="true"
          class="dropdown-item"
          disabled="true"
          type="button"
        >
          I'm a disabled button
        </button>
      </li>
      <li>
        <button
          class="dropdown-item active"
          type="button"
        >
          I'm a active button
        </button>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - dropdown renders correctly with dark mode 1`] = `
<jest>
  <div
    class="dropdown"
  >
    <button
      aria-expanded="false"
      autocomplete="off"
      class="dropdown-toggle btn btn-primary btn-md"
      data-bs-toggle="dropdown"
      id="dropdown-1"
      type="button"
    >
      Dropdown Toggle
    </button>
    <ul
      aria-labelledby="dropdown-1"
      class="dropdown-menu dropdown-menu-dark"
    >
      <li>
        <a
          class="dropdown-item"
          href="#"
        >
          I'm a link
        </a>
      </li>
      <li>
        <button
          class="dropdown-item"
          type="button"
        >
          I'm a button
        </button>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <button
          aria-disabled="true"
          class="dropdown-item"
          disabled="true"
          type="button"
        >
          I'm a disabled button
        </button>
      </li>
      <li>
        <button
          class="dropdown-item active"
          type="button"
        >
          I'm a active button
        </button>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - dropdown renders correctly with direction dropup 1`] = `
<jest>
  <div
    class="dropdown dropup"
  >
    <button
      aria-expanded="false"
      autocomplete="off"
      class="dropdown-toggle btn btn-primary btn-md"
      data-bs-toggle="dropdown"
      id="dropdown-1"
      type="button"
    >
      Dropdown Toggle
    </button>
    <ul
      aria-labelledby="dropdown-1"
      class="dropdown-menu"
    >
      <li>
        <a
          class="dropdown-item"
          href="#"
        >
          I'm a link
        </a>
      </li>
      <li>
        <button
          class="dropdown-item"
          type="button"
        >
          I'm a button
        </button>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <button
          aria-disabled="true"
          class="dropdown-item"
          disabled="true"
          type="button"
        >
          I'm a disabled button
        </button>
      </li>
      <li>
        <button
          class="dropdown-item active"
          type="button"
        >
          I'm a active button
        </button>
      </li>
    </ul>
  </div>
</jest>
`;

exports[`OE - dropdown renders correctly with trigger link 1`] = `
<jest>
  <div
    class="dropdown"
  >
    <a
      aria-expanded="false"
      class="dropdown-toggle"
      data-bs-toggle="dropdown"
      href="#"
      id="dropdown-1"
      role="button"
    >
      Dropdown Toggle
    </a>
    <ul
      aria-labelledby="dropdown-1"
      class="dropdown-menu"
    >
      <li>
        <a
          class="dropdown-item"
          href="#"
        >
          I'm a link
        </a>
      </li>
      <li>
        <button
          class="dropdown-item"
          type="button"
        >
          I'm a button
        </button>
      </li>
      <li>
        <hr
          class="dropdown-divider"
        />
      </li>
      <li>
        <button
          aria-disabled="true"
          class="dropdown-item"
          disabled="true"
          type="button"
        >
          I'm a disabled button
        </button>
      </li>
      <li>
        <button
          class="dropdown-item active"
          type="button"
        >
          I'm a active button
        </button>
      </li>
    </ul>
  </div>
</jest>
`;
